auth/src/main/java/com/google/cloud/auth/samples/DownscopingExample.java

```diff
@@ -0,0 +1,150 @@
+
+    // [START auth_downscoping_initialize_downscoped_cred]
+    // Create the downscoped credentials.
+    DownscopedCredentials downscopedCredentials =
+        DownscopedCredentials.newBuilder()
+            .setSourceCredential(sourceCredentials)
+            .setCredentialAccessBoundary(credentialAccessBoundary)
+            .build();
+
+    // Retrieve the token.
+    // This will need to be passed to the Token Consumer.
+    AccessToken accessToken = downscopedCredentials.refreshAccessToken();
+    // [END auth_downscoping_initialize_downscoped_cred]
+    return accessToken;
+  }
+  // [END auth_downscoping_token_broker]
+
+  /** Simulates token consumer readonly access to the specified object. */
+  // [START auth_downscoping_token_consumer]
+  public static String tokenConsumer(final String bucketName, final String objectName)
+      throws IOException {
+    // You can pass an `OAuth2RefreshHandler` to `OAuth2CredentialsWithRefresh` which will allow the
+    // library to seamlessly handle downscoped token refreshes on expiration.
+    OAuth2CredentialsWithRefresh.OAuth2RefreshHandler handler =
+        new OAuth2CredentialsWithRefresh.OAuth2RefreshHandler() {
+          @Override
+          public AccessToken refreshAccessToken() throws IOException {
+            // The common pattern of usage is to have a token broker pass the downscoped short-lived
+            // access tokens to a token consumer via some secure authenticated channel.
+            // For illustration purposes, we are generating the downscoped token locally.
+            // We want to test the ability to limit access to objects with a certain prefix string
+            // in the resource bucket. objectName.substring(0, 3) is the prefix here. This field is
+            // not required if access to all bucket resources are allowed. If access to limited
+            // resources in the bucket is needed, this mechanism can be used.
+            return getTokenFromBroker(bucketName, objectName.substring(0, 3));
+          }
+        };
+
+    // Downscoped token retrieved from token broker.
+    AccessToken downscopedToken = handler.refreshAccessToken();
+
+    // Create the OAuth2CredentialsWithRefresh from the downscoped token and pass a refresh handler
+    // which will handle token expiration.
+    // This will allow the consumer to seamlessly obtain new downscoped tokens on demand every time
+    // token expires.
+    OAuth2CredentialsWithRefresh credentials =
+        OAuth2CredentialsWithRefresh.newBuilder()
+            .setAccessToken(downscopedToken)
+            .setRefreshHandler(handler)
+            .build();
+
+    // Use the credentials with the Cloud Storage SDK.
+    StorageOptions options = StorageOptions.newBuilder().setCredentials(credentials).build();
+    Storage storage = options.getService();
+
+    // Call Cloud Storage APIs.
+    Blob blob = storage.get(bucketName, objectName);
+    return new String(blob.getContent());
```

Comment:
It would be nice to print here instead like: 
```
String content = new String(blob.getContent());
System.out.println("Retrieved object, " + objectName + ", from bucket," + bucketName + ", with content: " + content);
```
We usually test for a print out like this like `assertThat(output).contains("Retrieved object ... ")` and have this method return void.